### ISSUE TITLE
Can't use Put for dynamodb

### ISSUE THREAD

**Issue by whiteshadoww** [fogfish/dynamo#35]
ValidationException: One or more parameter values are not valid. The AttributeValue for a key attribute cannot contain an empty string value. Key: prefix

**Comment by fogfish** [comment]
Thank you for the report and apologise on the delay! 
The issue is caused either by missing the ID definition in the type or its value is not defined. In order to support hierarchical relations the library exposes a strict schema on the identity of entries.

Here is the example of the type definition

```go
type Person struct {
  dynamo.ID
  Name    string `dynamodbav:"name,omitempty"`
  Age     int    `dynamodbav:"age,omitempty"`
  Address string `dynamodbav:"address,omitempty"`
}
```

Could please share your type definition and code snippet about the instantiation of the type?

**Comment by fogfish** [comment]
closed by #38
